packages/peregrine/lib/talons/SearchBar/useAutocomplete.js

```diff
@@ -15,39 +15,48 @@ import debounce from 'lodash.debounce';
  */
 export const useAutocomplete = props => {
     const { query, visible } = props;
+    const [cleared, setCleared] = useState(false);
 
     // prepare to run query
     const [runQuery, queryResult] = useLazyQuery(query);
-    const { data, error, loading } = queryResult;
+    const { called, data, error, loading } = queryResult;
 
     // retrieve value from another field
     const { value } = useFieldState('search_query');
     const valid = value && value.length > 2;
 
     // Create a debounced function so we only search some delay after the last
     // keypress.
-    const debouncedRunQuery = useMemo(() => debounce(runQuery, 500), [
-        runQuery
-    ]);
+    const debouncedRunQuery = useMemo(
+        () =>
+            debounce(options => {
+                setCleared(false);
+                runQuery(options);
+            }, 500),
+        [runQuery, setCleared]
+    );
 
     // run the query once on mount, and again whenever state changes
     useEffect(() => {
         if (visible && valid) {
             debouncedRunQuery({ variables: { inputText: value } });
+        } else if (called && !value && !visible) {
+            setCleared(true);
```

Comment:
What's happening is that as you delete the text of `tops dresses` the last "valid" search text is `top` which is used as the debounced query. That explains why you see product suggestions for `top` which are based on the last search data, but you don't see any text in the `<input> in <Category>` suggestions.

Reply:
I fixed this by moving the `setCleared` logic into the debounced run query. The last call that executes should control the UI. In this case, we don't need to validate input before the debounce, we can do that inside the debounce, and if it is invalid we just clear the view.